testFlight1/lib/teensyShared/rocket.cpp

```diff
@@ -0,0 +1,244 @@
+                rocketTimers.coast_timer =
+                    chVTGetSystemTime() - rocketTimers.burnout_time;
+ 
+                // If the low acceleration lasts long enough, coast is detected
+                if (rocketTimers.coast_timer > coast_time_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_COAST;
+                }
+ 
+            break;
+ 
+            case STATE_COAST:
+                //if velocity <= 0
+                if (pointer_struct->sensorDataPointer->state_data.state_vz <= apogee_thresh) {
+                    rocketTimers.apogee_time = chVTGetSystemTime();
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_APOGEE_DETECT;
+                }
+ 
+            break;
+ 
+            case STATE_APOGEE_DETECT:
+                //If the negative velocity was too brief, go back to coast
+                if (pointer_struct->sensorDataPointer->state_data.state_vz > apogee_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_COAST;
+                    break;
+                }
+ 
+                // measure the length of the apogee time
+                rocketTimers.apogee_timer =
+                    chVTGetSystemTime() - rocketTimers.apogee_time;
+ 
+                // If the < 0 velocity lasts long enough, apogee is detected
+                if (rocketTimers.coast_timer > apogee_time_thresh) {
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_APOGEE;
+                }
+            break;
+ 
+            case STATE_APOGEE: //*assumes velocity is <= 0
+                //If rocket has velocity in main range
+                if (pointer_struct->sensorDataPointer->state_data.state_vz < main_thresh_max
+                        && pointer_struct->sensorDataPointer->state_data.state_vz > main_thresh_min) {
+                    rocketTimers.main_time = chVTGetSystemTime();
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_MAIN_DETECT;
+                }
+                //If rocket has volocity in drogue range
+                if (pointer_struct->sensorDataPointer->state_data.state_vz < drogue_thresh) {
+                    rocketTimers.drogue_time = chVTGetSystemTime();
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_DROGUE_DETECT;
+                }
+                //If rocket has 0ish velocity
+                if (pointer_struct->sensorDataPointer->state_data.state_vz > landed_thresh) {
+                    rocketTimers.landing_time = chVTGetSystemTime();
+                    pointer_struct->sensorDataPointer->rocketState_data.rocketState = STATE_LANDED_DETECT;
+                }
+            break;
+ 
+            case STATE_DROGUE_DETECT:
+                //If the acceleration was too brief, go back to apogee
```

Comment:
Talking about acceleration here and various locations below. Should be vertical velocity or descent rate.